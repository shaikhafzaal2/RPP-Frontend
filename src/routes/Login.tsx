import React from 'react';


 



// import { useDispatch, useSelector } from 'react-redux';
// import styled from 'styled-components';
// import { Button, Container, responsive, Text } from 'styled-minimal';

// import { spacer } from 'modules/theme';


import Background from 'components/Background';
import SplitScreen from 'components/SplitScreen';


function Login() {
 

  return (
    <Background key="Home" data-testid="Home">
      <SplitScreen />
      
    </Background>
  );
}


export default Login;

