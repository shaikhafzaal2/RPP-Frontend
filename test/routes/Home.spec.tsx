import React from 'react';
import Login from 'routes/Login';



import { fireEvent, render, screen } from 'test-utils';

const mockDispatch = jest.fn();

describe('Home', () => {
  it('should render properly', () => {
    render(<Login />);
    expect(screen.getByTestId('Home')).toMatchSnapshot();
  });

  // it('should handle clicks', () => {
  //   render(<Home />, { mockDispatch });
  //   fireEvent.click(screen.getByTestId('Login'));

  //   expect(mockDispatch).toHaveBeenCalledWith({
  //     type: 'USER_LOGIN_REQUEST',
  //   });
  // });
});
