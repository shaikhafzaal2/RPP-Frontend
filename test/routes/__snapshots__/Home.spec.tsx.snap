// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render properly 1`] = `
.c0 {
  background: #ffffff;
  color: #ffffff;
  min-height: 100vh;
  overflow: hidden;
  position: relative;
}

.c6 {
  width: 90%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  width: 60%;
  height: 300px;
  background-color: #FFFFFF;
  border: 2px solid #433270;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 8px;
}

.c8 {
  font-size: 24px;
  margin: 0;
  padding: 30px;
  background-color: #FFFFFF;
  color: #433270;
}

.c9 {
  width: 80%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  width: 100%;
  height: 40px;
  margin-bottom: 20px;
  padding: 10px;
  font-size: 16px;
  border-radius: 6px;
  border: 2px solid #433270;
}

.c11 {
  padding: 10px;
}

.c12 {
  width: 100%;
  height: 40px;
  background-color: #433270;
  color: #FFFFFF;
  font-size: 16px;
  cursor: pointer;
  border-radius: 6px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
}

.c2 {
  width: 50%;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  width: 60%;
  height: 25%;
  margin-bottom: 40px;
}

.c4 {
  font-family: 'Noto Sans',sans-serif;
  font-weight: 600;
  font-size: 40px;
  color: #C73E27;
  text-align: center;
  padding: 0 -20px;
}

.c5 {
  width: 50%;
  background-color: #FFFFFF;
}

.c13 {
  position: absolute;
  top: 150px;
  bottom: 199px;
  left: 50%;
  width: 2px;
  height: 408px;
  background-color: #c73e27;
}

<div
  class="c0"
  data-testid="Home"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <img
        alt="Image"
        class="c3"
        src="https://cdn.msruas.ac.in/ruas/imager/logos/identity/114151/university-main-logo_96575fd44b35071619e2d5f4abc0b108.png"
      />
      <span
        class="c4"
      >
        Welcome to 
      </span>
      <span
        class="c4"
      >
        PLACEMENT PORTAL
      </span>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <h1
            class="c8"
          >
            Sign In
          </h1>
          <form
            class="c9"
          >
            <select
              class="c10"
              name="role"
            >
              <option
                class="c11"
                value="admin"
              >
                Admin
              </option>
              <option
                class="c11"
                value="student"
              >
                Student
              </option>
            </select>
            <button
              class="c12"
              type="submit"
            >
              Sign In
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      class="c13"
    />
  </div>
</div>
`;
